src/main/java/dk/sdu/se_f22/brandmodule/index/BrandIndex.java

```diff
@@ -17,7 +21,34 @@
 
     @Override
     public void indexBrandInformation(Brand brand, List<String> tokens) {
-
+        try {
+            PreparedStatement mapInsert = DBConn.prepareStatement("INSERT INTO tokenbrandmap(brandid, tokenid) VALUES (?,?)");
+            PreparedStatement tokenInsert = DBConn.prepareStatement("INSERT INTO tokens(token) VALUES (?)");
+            PreparedStatement queryToken = DBConn.prepareStatement("SELECT token FROM tokens");
+
+            ResultSet rs = queryToken.executeQuery();
+
+            ArrayList<String> newTokens = new ArrayList<>(tokens);
+
+            for (int i = 0; i < tokens.size(); i++){
+                while (rs.next()){
+                    if (rs.getString("token") != newTokens.get(i)){
+                        tokenInsert.setString(1, newTokens.get(i));
+                    }
+                }
+                PreparedStatement queryTokenId = DBConn.prepareStatement("SELECT id FROM tokens where token = ?");
```

Comment:
Your function is doing a lot of different things. for instance preparing 4 different statements. 
Wouldn't it make sense to abstract your query handling away from the public function.
This would:

- make the code more readable
- Allow you to have different Error messages in your catch blocks
- Keep the level of abstraction identical in your function

Reply:
We don't quite understand what you mean by this, could you give us an example.